### ISSUE TITLE
Unable to use mongo _id for await User.find(_id). 

### ISSUE THREAD

**Issue by borsemayur2** [eveningkid/denodb#234]
Able to find user with _id (objectID2) and user generated _id (abc2) but not working with auto-generated _id ("607cf59ec294c8c3f1261e4f")
```
await User.find(_id as any)  // works with _id as objectId
User { 
  _id: ObjectId2 {
    [Symbol(id)]: Buffer2(12) [       
       96, 124, 245, 158, 194,        
      148, 200, 195, 241,  38,        
       30,  79
    ]
  },
  name: "user1"
}

await User.find("607cf59ec294c8c3f1261e4f")  // undefined

await User.find("abc1") // works
```

Auto-generated _id is `607cf59ec294c8c3f1261e4f`
![image](https://user-images.githubusercontent.com/47341981/115177164-424cfd80-a0ec-11eb-97ca-4bd804e66f2f.png)

Auto-generated _id returns undefined
![image](https://user-images.githubusercontent.com/47341981/115177212-5ee93580-a0ec-11eb-88b2-0c670c345024.png)

User generated _id works
![image](https://user-images.githubusercontent.com/47341981/115177312-90fa9780-a0ec-11eb-96e7-3b74d1a271ec.png)

**Comment by borsemayur2** [comment]
I think we need to use following for converting `_id` `string` to  `objectId`

```deno
import { Bson } from "https://deno.land/x/mongo@v0.22.0/mod.ts";

const id = "abc123xyz";
const _id: any = new Bson.ObjectId(id);

await User.find(id);  // undefined
await User.find(_id); // works
```

**Comment by borsemayur2** [comment]
> 
> 
> I think we need to use following for converting `_id` `string` to `objectId`
> 
> ```
> import { Bson } from "https://deno.land/x/mongo@v0.22.0/mod.ts";
> 
> const id = "abc123xyz";
> const _id: any = new Bson.ObjectId(id);
> 
> await User.find(id);  // undefined
> await User.find(_id); // works
> ```

I'd like to create `pr` for this if this seems correct.

**Comment by borsemayur2** [comment]
I've tried this approach in #239

**Comment by eveningkid** [comment]
Hey Mayur,

Thank you for sharing all this. I'm sure this can easily be fixed :)

I've looked at the mongo connector and basically whenever you do `.find(...)` this is what it does:

```js
await User.find(id);
// =>
await User.where({ _id: id }).get();
```

If we make an update, delete something, the id problem would be similar:

```js
await User.where({ _id: id }).update({ name: "Mayur" });
```

From my perspective, it then seems that the `where` clause is the issue every time.

What we could do is to check whenever the connector has to deal with `where`s, it could:
- Check for the field name
- If it equals to `_id`, then wrap it with an object id
- Otherwise do nothing (we keep the field name)

I could give it a try but if you're adventurous:
- Look for `.wheres` and `.whereIn` into the [mongo connector](https://github.com/eveningkid/denodb/blob/master/lib/connectors/mongodb-connector.ts)
- [This is how `.wheres` are added to a query](https://github.com/eveningkid/denodb/blob/8dfb7ad30862b9d5a9bed967cda6c48a0101946a/lib/query-builder.ts#L177)
- [This is how `.whereIn` are added to a query](https://github.com/eveningkid/denodb/blob/8dfb7ad30862b9d5a9bed967cda6c48a0101946a/lib/query-builder.ts#L142)

The goal is to only update the mongo connector, not `Model`  that should remain generic.

Let me know if you want to give it a try :)

**Comment by borsemayur2** [comment]
Sure. I'll try it.

**Comment by borsemayur2** [comment]
> 
> 
> Hey Mayur,
> 
> Thank you for sharing all this. I'm sure this can easily be fixed :)
> 
> I've looked at the mongo connector and basically whenever you do `.find(...)` this is what it does:
> 
> ```js
> await User.find(id);
> // =>
> await User.where({ _id: id }).get();
> ```
> 
> If we make an update, delete something, the id problem would be similar:
> 
> ```js
> await User.where({ _id: id }).update({ name: "Mayur" });
> ```
> 
> From my perspective, it then seems that the `where` clause is the issue every time.
> 
> What we could do is to check whenever the connector has to deal with `where`s, it could:
> 
>     * Check for the field name
> 
>     * If it equals to `_id`, then wrap it with an object id
> 
>     * Otherwise do nothing (we keep the field name)
> 
> 
> I could give it a try but if you're adventurous:
> 
>     * Look for `.wheres` and `.whereIn` into the [mongo connector](https://github.com/eveningkid/denodb/blob/master/lib/connectors/mongodb-connector.ts)
> 
>     * [This is how `.wheres` are added to a query](https://github.com/eveningkid/denodb/blob/8dfb7ad30862b9d5a9bed967cda6c48a0101946a/lib/query-builder.ts#L177)
> 
>     * [This is how `.whereIn` are added to a query](https://github.com/eveningkid/denodb/blob/8dfb7ad30862b9d5a9bed967cda6c48a0101946a/lib/query-builder.ts#L142)
> 
> 
> The goal is to only update the mongo connector, not `Model` that should remain generic.
> 
> Let me know if you want to give it a try :)

Tried this in #240

**Comment by eveningkid** [comment]
This should now be fixed thanks to @borsemayur2 work in #240 :)

The changes have been added to `1.0.35`.

Have a great day!
